crates/iceberg/src/spec/snapshot.rs

```diff
@@ -408,9 +408,19 @@ pub(super) mod _serde {
                 timestamp_ms: v1.timestamp_ms,
                 manifest_list: match (v1.manifest_list, v1.manifests) {
                     (Some(file), None) => file,
-                    (Some(_), Some(_)) => "Invalid v1 snapshot, when manifest list provided, manifest files should be omitted".to_string(),
-                    (None, _) => "Unsupported v1 snapshot, only manifest list is supported".to_string()
-                   },
+                    (Some(_), Some(_)) => {
+                        return Err(Error::new(
+                            ErrorKind::Unexpected,
+                            "v1 snapshot invariant violated: manifest_list and manifests are both set",
```

Comment:
```suggestion
                            "Invalid v1 snapshot, when manifest list provided, manifest files should be omitted",
```
I think the original error message is more specific